get_mvn.go

```diff
@@ -0,0 +1,161 @@
+package getter
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// MvnGetter is a Getter implementation that will download an artifact from maven repository, e.g. Sonatype Nexus,
+// uri format: mvn::http://[username@]hostname[:port]/directoryname[?options]
+type MvnGetter struct {
+	HttpGet HttpGetter
+}
+
+func (g *MvnGetter) ClientMode(u *url.URL) (ClientMode, error) {
+	return ClientModeFile, nil
+}
+
+func (g *MvnGetter) Get(dst string, u *url.URL) error {
+	q := u.Query()
+	artifactId := q.Get("artifactId")
+	if artifactId == "" {
+		return fmt.Errorf("query parameter 'artifactId' is required.")
+	}
+	version := q.Get("version")
+	if version == "" {
+		return fmt.Errorf("query parameter 'version' is required.")
+	}
+	classifier := q.Get("classifier")
+
+	artType := q.Get("type")
+	if artType == "" {
+		artType = "jar"
+	}
+
+	filename := artifactId + "-" + version
+	if classifier != "" {
+		filename += "-" + classifier
+	}
+	filename += "." + artType
```

Comment:
`filename` is the name of the file, not relate to file path, so string concatenation is OK here.

Reply:
It's constructed out of arbitrary query parameters and appended to a path, so I think we should clean it.

For example if the URL contains something like `?artifactId=../foo` it would get joined to `dstDir` and end up escaping `dstDir`.

This assumes the Nomad situation where `dstDir` is controlled by an application (Nomad) but the URL is accepted as user input. Since the URL is user input we need to make sure it doesn't escape `dstDir`.

Reply:
ok, i see, thanks for the detail explanation